src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.Windows.cs

```diff
@@ -54,7 +54,7 @@ private TarEntry ConstructEntryForWriting(string fullPath, string entryName, Fil
             entry._header._aTime = info.LastAccessTimeUtc;
             entry._header._cTime = info.LastWriteTimeUtc; // There is no "change time" property
 
-            entry.Mode = DefaultWindowsMode;
+            entry.Mode = attributes.HasFlag(FileAttributes.Directory) ? TarHelpers.DefaultDirectoryMode : WindowsFileMode;
```

Comment:
@carlossanlop I've made this change so that across all platforms there are no default permissions that have the group/other write bit set.